signalfx/ingest.py

```diff
@@ -271,31 +273,55 @@ def _set_event_properties(self, pbuf_obj, properties):
-        if isinstance(value, int):
-            prop.value.intValue = value
-        elif isinstance(value, str):
-            prop.value.strValue = value
-        elif isinstance(value, unicode):
-            prop.value.strValue = value
-        elif isinstance(value, float):
-            prop.value.doubleValue = value
-        elif isinstance(value, bool):
-            prop.value.boolValue = value
-        else:
-            raise ValueError('Invalid Value ' + str(value))
-
-    def _assign_value_type(self, pbuf_dp, value):
-        if isinstance(value, int):
-            pbuf_dp.value.intValue = value
-        elif isinstance(value, str):
-            pbuf_dp.value.strValue = value
-        elif isinstance(value, unicode):
-            pbuf_dp.value.strValue = value
+    def _assign_value_by_type(self, pbuf_obj, value, _bool=True, _float=True,
+                        _integer=True, _string=True):
+        """Assigns the supplied value to the appropriate protobuf value type"""
+        # bool inherits int, so bool instance check must be executed prior to
+        # checking for integer types
+        if isinstance(value, bool):
+            if _bool is True:
+                pbuf_obj.value.boolValue = value
+            else:
+                raise ValueError('boolean types are not supported')
+        elif isinstance(value, six.integer_types):
+            if _integer is True:
+                if value > INTEGER_MAX:
+                    raise ValueError(str(value) + ' exceeds maximum signed '
+                                     + '64 bit integer as defined by protobuf'
+                                     + ' (' + str(INTEGER_MAX) + ')')
+                if value < INTEGER_MIN:
+                    raise ValueError(str(value) + ' less than minimum signed '
+                                     + '64 bit integer as defined by protobuf '
+                                     + ' (' + str(INTEGER_MIN) + ')')
+                pbuf_obj.value.intValue = value
+            else:
+                raise ValueError('integer and long types are not supported')
         elif isinstance(value, float):
-            pbuf_dp.value.doubleValue = value
+            if _float is True:
+                pbuf_obj.value.doubleValue = value
+            else:
+                raise ValueError('float types are not supported')
+        elif isinstance(value, six.string_types):
+            if _string is True:
+                pbuf_obj.value.strValue = value
+            else:
+                raise ValueError('string and unicode types are not supported')
         else:
-            raise ValueError('Invalid Value ' + str(value))
+            raise ValueError(str(value))
+
+    def _assign_property_value(self, prop, value):
+        """Assigns a property value to the protobuf obj property"""
+        try:
+            self._assign_value_by_type(prop, value)
+        except ValueError as e:
+            raise ValueError('Invalid property value ' + str(e))
```

Comment:
Agreed, but combining the two functions took away the ability to provide specific messages.  Since there is a distinct difference from property values, and actual values I think that distinction still needs to be made.  That way a developer less likely to walk through our code to find out where their offense occurred.